src/transformers/models/dinat/modeling_dinat.py

```diff
@@ -871,3 +873,122 @@ def forward(
             attentions=outputs.attentions,
             reshaped_hidden_states=outputs.reshaped_hidden_states,
         )
+
+
+@add_start_docstrings(
+    "NAT backbone, to be used with frameworks like DETR and MaskFormer.",
+    DINAT_START_DOCSTRING,
+)
+class DinatBackbone(DinatPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+
+        requires_backends(self, ["natten"])
+
+        self.stage_names = config.stage_names
+
+        self.embeddings = DinatEmbeddings(config)
+        self.encoder = DinatEncoder(config, self.embeddings.patch_grid)
+
+        self.out_features = config.out_features
```

Comment:
The dinat config does not have `out_features` or `stage_names`.

Reply:
Fixed now, thanks!